node-data/src/message.rs

```diff
@@ -872,25 +876,135 @@ pub mod payload {
         }
     }
 
-    #[derive(Default, Debug, Clone)]
-    pub struct GetData {
-        pub inner: Inv,
+    #[derive(Debug, Clone)]
+    pub struct GetResource {
+        /// Inventory/Resource to search for
+        inventory: Inv,
+
+        /// (requester) Address to which the resource is sent back, if found
+        requester_addr: SocketAddr,
+
+        /// Request Time to live
+        ttl_as_sec: u64,
```

Comment:
It's a valid suggestion. 
As a side node, we use a redundancy factor (K_BETA=3) in Kadcast per a bucket. That's said, limiting the number of hops should take this factor into account.

@herr-seppia  can also comment on that.

Reply:
TTL might be beneficial in the scenario of requesting the (network) blockchain tip.

If a block is not delivered for a period of 10s (or any value close to the expected average block time) then the requester most probably will need to request the new blockchain tip instead.

Reply:
Would it be feasible to implement both options? So that when requesting time-bound resources, like the tip, we could set a timestamp TTL, while when requesting less urgent resources we could use a counter.

Reply:
Yes, it is feasible indeed.

Reply:
On a second thought, the issue of "out of sync" clocks would be the same (two nodes might be 10 seconds apart, and the message would be dropped). 
We could handle the urgency of the "tip" case by setting the TTL counter to low values, so as to limit the search to just a few hops.
So I'd still go for the counter, after all.

Reply:
Both `GetResource::hops_limit` and `GetResource::TTL`  are now supported. In case of requesting a block, it uses `hops_limit=100` currently.